R/list-combine.R

```diff
@@ -40,7 +40,17 @@
 list_c <- function(x, ..., ptype = NULL) {
   vec_check_list(x)
   check_dots_empty()
-  list_unchop(x, ptype = ptype, error_call = current_env())
+
+  # TODO: Ideally we'd set `name_spec = zap()` to drop names from both `args`
+  # and its elements for performance, but that doesn't work for non-vctrs types.
+  # https://github.com/r-lib/vctrs/issues/1106
```

Comment:
```suggestion
  # For `list_c()`, we don't expose `list_unchop()`'s `name_spec` arg,
  # and instead strip outer names to avoid collisions with inner names
```

Reply:
I don't think the message is exactly the same, because we don't want to zap _all_ the names, just the outer ones